registry/handlers/hooks.go

```diff
@@ -0,0 +1,103 @@
+package handlers
+
+import (
+	"fmt"
+	"net/smtp"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/distribution/configuration"
+)
+
+// ReportHook is for hooking Panic in web application
+type ReportHook struct {
+}
+
+// Fire forwards an error to ReportHook
+func (hook *ReportHook) Fire(entry *logrus.Entry) error {
+	logrus.Info("Message: ", entry.Message)
+	if errConfig.Log.Handlers.Errors.Type == "email" {
+		if err := mailSending(errConfig, entry.Message); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Levels contains hook levels to be catched
+func (hook *ReportHook) Levels() []logrus.Level {
+	return []logrus.Level{
+		logrus.DebugLevel,
+		logrus.InfoLevel,
+		logrus.WarnLevel,
+		logrus.ErrorLevel,
+		logrus.FatalLevel,
+		logrus.PanicLevel,
+	}
+}
+
+var reportLevel = map[string]logrus.Level{
+	"debug": logrus.DebugLevel,
+	"info":  logrus.InfoLevel,
+	"warn":  logrus.WarnLevel,
+	"error": logrus.ErrorLevel,
+	"fatal": logrus.FatalLevel,
+	"panic": logrus.PanicLevel,
+}
+
+var errConfig = &configuration.Configuration{}
+
+// ErrorReport is for using message with hook, ex. ErrorReport.Panic
+var ErrorReport = logrus.New()
+
+// ConfigReportHook should be done when ReportHook is needed
+func ConfigReportHook(configuration *configuration.Configuration) {
+	errConfig = configuration
+	logrus.Info("Error Level: ", errConfig.Log.Handlers.Errors.Level)
+	hook := &ReportHook{}
+	ErrorReport.Hooks.Add(hook)
+	ErrorReport.Formatter = new(logrus.TextFormatter)
+	ErrorReport.Level = reportLevel[errConfig.Log.Handlers.Errors.Level]
+}
+
+func mailSending(configuration *configuration.Configuration, message string) error {
```

Comment:
Let's also think about supporting a template for the body. It doesn't have to be configurable for now, but it will provide the option in the future. There is a lot of information in the logger that could be expanded in an email template.